services/client.go

```diff
@@ -0,0 +1,135 @@
+package services
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"sync"
+	"time"
+
+	"github.com/frain-dev/convoy/datastore"
+	"github.com/gorilla/websocket"
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	// Maximum message size allowed from peer.
+	maxMessageSize = 512
+
+	maxDeviceLastSeenDuration = 2 * time.Minute
+)
+
+// Client is a middleman between the websocket connection and the hub.
+type Client struct {
```

Comment:
I think we can refactor this to be clearer:

- This type doesn't need a reference to the `hub`, the `hub` holds a reference to it already. 
- We can move this to `internal/pkg` and test it in isolation.